lucille-core/src/main/java/com/kmwllc/lucille/connector/storageclient/LocalStorageClient.java

```diff
@@ -23,130 +22,85 @@
-    try {
-      return pathToDoc(path, params);
-    } catch (ConnectorException e) {
-      throw new IllegalArgumentException("Unable to convert path '" + path + "' to Document", e);
+    @Override
+    public String getFullPath(TraversalParams params) {
+      return path.toString();
     }
-  }
 
-  @Override
-  protected Document convertFileReferenceToDoc(FileReference fileReference, InputStream in, String decompressedFullPathStr,
-      TraversalParams params) {
-    Path path = fileReference.getPath();
-    try {
-      return pathToDoc(path, in, decompressedFullPathStr, params);
-    } catch (ConnectorException e) {
-      throw new IllegalArgumentException("Unable to convert path '" + path + "' to Document", e);
+    @Override
+    public boolean isCloudFileReference() {
+      return false;
     }
-  }
 
-  @Override
-  protected InputStream getFileReferenceContentStream(FileReference fileReference, TraversalParams params) {
-    Path path = fileReference.getPath();
-    try {
-      return Files.newInputStream(path);
-    } catch (IOException e) {
-      throw new IllegalArgumentException("Unable to get content stream of path '" + path + "'", e);
+    @Override
+    public boolean isValidFile() {
+      return true;
     }
-  }
-
-  @Override
-  protected String getStartingDirectory(TraversalParams params) {
-    return params.getURI().getPath();
-  }
-
-  @Override
-  protected String getBucketOrContainerName(TraversalParams params) {
-    return params.getURI().getAuthority();
-  }
-
-  private Document pathToDoc(Path path, TraversalParams params) throws ConnectorException {
-    String fullPath = path.toAbsolutePath().normalize().toString();
-    String docId = DigestUtils.md5Hex(fullPath);
-    Document doc = Document.create(createDocId(docId, params));
-
-    try {
-      BasicFileAttributes attrs = Files.readAttributes(path, BasicFileAttributes.class);
 
-      doc.setField(FILE_PATH, fullPath);
-      doc.setField(SIZE, attrs.size());
-
-      if (attrs.lastModifiedTime() != null) {
-        doc.setField(MODIFIED, attrs.lastModifiedTime().toInstant());
-      }
-
-      if (attrs.creationTime() != null) {
-        doc.setField(CREATED, attrs.creationTime().toInstant());
+    @Override
+    public InputStream getContentStream(TraversalParams params) {
+      try {
+        return Files.newInputStream(path);
+      } catch (IOException e) {
+        throw new IllegalArgumentException("Unable to get content stream of path '" + path + "'", e);
       }
+    }
 
-      if (params.shouldGetFileContent()) {
-        doc.setField(CONTENT, Files.readAllBytes(path));
+    @Override
+    public Document toDoc(TraversalParams params) {
+      String fullPath = path.toAbsolutePath().normalize().toString();
+      String docId = DigestUtils.md5Hex(fullPath);
+      Document doc = Document.create(createDocId(docId, params));
+
+      try {
+        BasicFileAttributes attrs = Files.readAttributes(path, BasicFileAttributes.class);
```

Comment:
let's confirm that this isn't adding a file read where we weren't always doing it before